NDTensors/src/lib/Sectors/src/abstractcategory.jl

```diff
@@ -1,45 +1,130 @@
+  return GradedAxes.gradedrange(LabelledNumbers.LabelledInteger.(degen, C.(labels)))
 end
 
-function trivial(category_type::Type{<:AbstractCategory})
-  return error("`trivial` not defined for type $(category_type).")
+# abelian case: return Category
+function fusion_rule(::AbelianGroup, c1::C, c2::C) where {C<:AbstractCategory}
+  return C(label_fusion_rule(C, category_label(c1), category_label(c2)))
 end
 
-istrivial(c::AbstractCategory) = (c == trivial(typeof(c)))
+function label_fusion_rule(category_type::Type{<:AbstractCategory}, l1, l2)
+  return error("`label_fusion_rule` not defined for type $(category_type).")
+end
+
+# convenient to define fusion rule for LabelledInteger too
+# TBD expose this through ⊗? Currently not accessible.
+function fusion_rule(
+  ::SymmetryStyle, l1::LabelledNumbers.LabelledInteger, l2::LabelledNumbers.LabelledInteger
+)
+  blocks12 = BlockArrays.blocklengths(LabelledNumbers.label(l1) ⊗ LabelledNumbers.label(l2))
+  v =
+    LabelledNumbers.LabelledInteger.(l1 * l2 .* blocks12, LabelledNumbers.label.(blocks12))
+  return GradedAxes.gradedrange(v)
+end
+
+function fusion_rule(
+  ::AbelianGroup, l1::LabelledNumbers.LabelledInteger, l2::LabelledNumbers.LabelledInteger
+)
+  fused = LabelledNumbers.label(l1) ⊗ LabelledNumbers.label(l2)
+  return LabelledNumbers.LabelledInteger(l1 * l2, fused)
+end
+
+# =============  fusion rule and gradedunitrange ===================
+# GradedAxes.tensor_product interface. Only for abelian groups.
+function GradedAxes.fuse_labels(c1::AbstractCategory, c2::AbstractCategory)
+  return GradedAxes.fuse_labels(
+    combine_styles(SymmetryStyle(c1), SymmetryStyle(c2)), c1, c2
+  )
+end
+function GradedAxes.fuse_labels(::SymmetryStyle, c1::AbstractCategory, c2::AbstractCategory)
+  return error("`fuse_labels` is only defined for abelian groups")
+end
+function GradedAxes.fuse_labels(::AbelianGroup, c1::AbstractCategory, c2::AbstractCategory)
+  return fusion_rule(c1, c2)
+end
 
-function dual(category_type::Type{<:AbstractCategory})
-  return error("`dual` not defined for type $(category_type).")
+# cast to range
+to_graded_axis(c::AbstractCategory) = GradedAxes.gradedrange([c => 1])
+to_graded_axis(l::LabelledNumbers.LabelledInteger) = GradedAxes.gradedrange([l])
+to_graded_axis(g::AbstractUnitRange) = g
+
+# allow to fuse a category with a GradedUnitRange
+function GradedAxes.fusion_product(a, b)
+  return GradedAxes.fusion_product(to_graded_axis(a), to_graded_axis(b))
 end
 
-Base.isless(c1::AbstractCategory, c2::AbstractCategory) = isless(label(c1), label(c2))
+# fusion_product with one input to be used in generic fusion_product(Tuple...)
+# TBD define fusion_product() = gradedrange([sector(())=>1])?
+GradedAxes.fusion_product(x) = GradedAxes.fusion_product(to_graded_axis(x))
+
+# product with trivial = easy handling of UnitRangeDual + sort and merge blocks
+GradedAxes.fusion_product(g::AbstractUnitRange) = GradedAxes.fusion_product(trivial(g), g)
```

Comment:
Related to the comment below about the design of `GradedAxes.fusion_product(g1::BlockedUnitRange, g2::BlockedUnitRange)`, can this instead be implemented as: `GradedAxes.fusion_product(g::AbstractUnitRange) = blockmergesort(g)`? I think that would make it clearer to readers of this code what it is actually doing.

Reply:
I think it is best to preserve `dual` in `blockmergesort`, I implemented the rest.